mypy/newsemanal/semanal.py

```diff
@@ -2563,6 +2563,9 @@ def analyze_member_lvalue(self, lval: MemberExpr, explicit_type: bool, is_final:
             if cur_node and is_final:
                 # Overrides will be checked in type checker.
                 self.fail("Cannot redefine an existing name as final", lval)
+            if (not lval.node and cur_node and isinstance(cur_node.node, Var) and
+                    cur_node.node.is_inferred and explicit_type):
+                self.name_already_defined('self.' + lval.name, lval, cur_node)
```

Comment:
Using this in the new analyzer is dangerous, it not not only shows the error, but also has some special code for deferring unknown names.

Reply:
I couldn't understand what you meant. Could you explain it a little more?

Reply:
I actually confused with with `name_not_defined()`. This one is OK. Anyway, did you check that in the situation where an error is emitted, no _new_ variable created, and the l.h.s. points to an existing one?

Reply:
yes, basically if `cur_node` is not `None`, l.h.s. points to an existing variable.